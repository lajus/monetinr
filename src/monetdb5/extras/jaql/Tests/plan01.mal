jaql.x("plan a -> transform { \"a\":$.j };");
jaql.x("plan a -> transform { \"a\":$.j[1] };");
jaql.x("plan a -> transform { \"a\":$.j[*].k };");

jaql.x("plan a -> transform { \"a\":$.* };");
catch MALException:str;
	io.printf("!%s\n", MALException);
exit MALException;
jaql.x("plan a -> transform { $.* };");
jaql.x("plan a -> transform { $.*.foo };");
catch MALException:str;
	io.printf("!%s\n", MALException);
exit MALException;
