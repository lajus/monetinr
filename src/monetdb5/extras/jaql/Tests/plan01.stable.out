stdout of test 'plan01` in directory 'monetdb5/extras/jaql` itself:


# 15:26:53 >  
# 15:26:53 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/net/sofia.ins.cwi.nl/export/scratch1/fabian/tmp/mtest-jacqueline-sofia.ins.cwi.nl/five/dbfarm" "--set" "mapi_open=true" "--set" "mapi_port=37247" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_extras_jaql" "plan01.mal"
# 15:26:53 >  

# MonetDB 5 server v11.8.0 "jacqueline-480797554eb5"
# Serving database 'mTests_extras_jaql', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.630 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2013 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://sofia.ins.cwi.nl:37247/
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/DataCell module not loaded: MALException:jaql.context:JAQL environment not found
function user.main():void;
    jaql.x("plan a -> transform { \"a\":$.j };");
    jaql.x("plan a -> transform { \"a\":$.j[1] };");
    jaql.x("plan a -> transform { \"a\":$.j[*].k };");
    jaql.x("plan a -> transform { \"a\":$.* };");
catch MALException:str ;
    io.printf("!%s\n",MALException);
exit MALException:str ;
    jaql.x("plan a -> transform { $.* };");
    jaql.x("plan a -> transform { $.*.foo };");
catch MALException:str ;
    io.printf("!%s\n",MALException);
exit MALException:str ;
end main;
a as $ -> transform: { "a": $.j } 
a as $ -> transform: { "a": $.j[1] } 
a as $ -> transform: { "a": $.j[*].k } 
!MALException:jaql.execute:syntax error, unexpected '*', expecting IDENT at or around '...form { "a":$.* };'
a as $ -> transform: { <to be deduced from expansion> $.* } 
!MALException:jaql.execute:syntax error, unexpected '.', expecting ',' or '}' at or around '...ansform { $.*.foo };'


# 15:26:53 >  
# 15:26:53 >  "Done."
# 15:26:53 >  

